### PR TITLE
Move all documentation to docs/

### PR DESCRIPTION
**Pull request by zdykstra**
The project root is getting a bit busy. Lets move anything related to documentation (excluding README.md) to `docs/`. All scripts/tools that consume the various documentation sources have been updated to reflect the new path.

### CODE REVIEW

**Review by ahesford — APPROVED** [review]
I haven't checked for any overlooked links in our documentation, but this looks good to me.

**Comment by zdykstra** [comment]
Thanks - fixed the links in `README.md` for `BOOTENVS.md` and `BUILD.md`.  I think that's all of them, but I'll check again with fresh eyes before I smash the merge button.

**Comment by zdykstra** [comment]
I think I caught it all. Merging!

**Pull request merged**
